delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/plugin/Plugin.scala

```diff
@@ -0,0 +1,31 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.plugin
+
+import akka.http.scaladsl.server.Route
+import monix.bio.Task
+
+/**
+  * Plugin API.
+  */
+trait Plugin {
+
+  /**
+    * Start the plugin.
+    *
+    * @param deps  plugin dependencies
+    * @return optional Akka Http [[Route]], if the plugin defines one.
+    */
+  def start(deps: PluginDependencies): Task[Option[Route]]
```

Comment:
I think we need the opportunity for a plugin to provide (register) contract definitions and instances during startup (such that plugins could possibly depend on other plugins).
A plugin also needs to be able to lookup instances provided by other plugins.

Reply:
So maybe the plugin interface should be split in two, like so:

```
trait PluginDef {
  def name: Name
  def version: String
  def dependencies: Set[PluginDef]
  def initialise(registry: Registry): Task[Plugin]
}
trait Plugin {
  def name: Name
  def version: String
  def dependencies: Set[PluginDef]

  def route: Route
  def stop(): Task[Unit]
}
trait Registry {
  def register[A](value: A): Task[Unit]
  def lookup[A]: IO[ReferenceNotFound, A]
}
```
It would allow declaring dependencies before initialisation, an opportunity to register instances during initialisation, route registration after initialisation and a stop hint for coordinated shutdown.

We could simplify things by dropping the status for now... we could simply expect that plugins manage their own lifecycle; during the service shutdown, we can notify plugins to stop whatever they're doing.